### PR TITLE
fix warning when logging in on tty (variable not set)

### PR DESCRIPTION
**Pull request by drm00**
When logging in on the TTY on OpenBSD, the variable $terminfo[colors] is not set, which results in an error while comparing with 8.

Since $terminfo is a read-only variable, i had to take care of the value extraction first before comparing the results, using 0 as the default value if $terminfo[colors] is unset.

### CODE REVIEW

File: modules/utility/init.zsh [hunk 1]
```diff
@@ -6,7 +6,8 @@
 # Colours
 #
 
-if (( ${terminfo[colors]} >= 8 )); then
```
Comment (Eriner) [inline modules/utility/init.zsh]: I believe you can do this inline, `${terminfo[colors]:=0}`

Comment (drm00) [inline modules/utility/init.zsh]: I tried it like this first, but if $terminfo[colors] is empty, zsh tells me that i can't change a readonly variable, like in this example:

`
$ readonly TEST
$ echo $TEST

$ echo ${TEST:=0}
zsh: read-only variable: TEST
`
Sadly i see no way around using a temporary variable :(

Comment (Eriner) [inline modules/utility/init.zsh]: In that case, perhaps:

 `if [[ ! -z ${terminfo[colors]} ]] && (( ${terminfo[colors]} >= 8 )); then`

I realize that the `! -z` is a double negative in reality, but I think it makes it more clear *why* we're checking it.

Doing that, it should first ensure that the key-value pair exists, and if it does, then verify that colors >= 8.

Comment (drm00) [inline modules/utility/init.zsh]: This works, just tried it out. Thanks for finding a better solution ;)

Comment (Eriner) [inline modules/utility/init.zsh]: @drm00 nope, this was all you, just recommending a small syntax change. If you update your commit, I'll merge this into master. Thanks!

Comment (drm00) [inline modules/utility/init.zsh]: Should be good now. Thanks!

**Comment by ericbn** [comment]
I've simplified the expression from

    if [[ ! -z ${terminfo[colors]} ]] && (( ${terminfo[colors]} >= 8 )); then

to

    if (( terminfo[colors] >= 8 )); then

at https://github.com/Eriner/zim/commit/6022fe4.

This is preferred and avoids variable substitution according to http://www.bash2zsh.com/zsh_refcard/refcard.pdf

> `var` (does not require `$` in front unless some substitution e.g. `${#var}` is needed, `$` is error if `var` is to be modified)

Without substitution, the undefined variable is defaulted to 0:

```
$ (( ${no_such_array[foo]} >= 8 )) && echo Hi
zsh: bad math expression: operand expected at `>= 8 '
$ (( no_such_array[foo] >= 8 )) && echo Hi
$ (( no_such_array[foo] == 0 )) && echo Hi
Hi
$ typeset -A no_such_array
$ no_such_array=( foo 42 )
$ (( no_such_array[foo] >= 8 )) && echo Hi
Hi
$
```

@Eriner, why did you consider this a bug at https://github.com/Eriner/zim/commit/abbeeb9? The changes there bring this issue back.

**Comment by ericbn** [comment]
I get the same behaviour as above in
* zsh 5.4.2 (x86_64-apple-darwin15.6.0) *macOS*
* zsh 5.0.8 (x86_64-apple-darwin15.0) *macOS*
* zsh 5.4.2 (arm-unknown-linux-androideabi) *Termux*
* zsh 5.1.1 (x86_64-ubuntu-linux-gnu) *Ubuntu on Windows Subsystem for Linux*

**Pull request closed without merging**